### PR TITLE
Add a failing unittest case when using `--pdb` option

### PR DESCRIPTION
**Pull request by asfaltboy**
Here's a failing test for #405, tests on travis-ci and tox pass. This helped to isolate it to occurring on latest `pytest==3.0.3`, probably has something to do with the `pdb` changes in the last minor releases. Specifically, [changelog for 3.0.2](https://github.com/pytest-dev/pytest/blob/master/CHANGELOG.rst#302) includes merge of PR pytest-dev/pytest#1890:

> Do not call tearDown and cleanups when running tests from unittest.TestCase subclasses with --pdb enabled. This allows proper post mortem debugging for all applications which have significant logic in their tearDown machinery (#1890). Thanks @mbyt for the PR.

In essence, it's not a bad change; but, when combined with `--reuse-db` or when `setUp` create unique Django objects this fails hard.

I guess one way, would be to re-enable this cleanup for `pytest-django` only.

But, to be honest, not sure how many more devs may spend hours looking for their missing tearDown code. On the other hand, I myself have suffered from ["early cleanUp when using pdb"](https://github.com/pytest-dev/pytest-mock/pull/58#issuecomment-253556301). Hmmm, maybe there is a better way.

Other users have noted this can cause issues at pytest-dev/pytest#1932

### CODE REVIEW

**Comment by asfaltboy** [comment]
Tests on travis-ci and tox pass, so I managed to isolate it to occurring on latest `pytest==3.0.3`, probably has something to do with the `pdb` changes in the last minor releases.

**Comment by blueyed** [comment]
Can you merge/cherry-pick https://github.com/pytest-dev/pytest-django/pull/400 into this, so it fails here, too, please?

**Comment by asfaltboy** [comment]
Added a quick workaround in latest commit d5696a2 ; this is basically copy-pasta of `django.test.testcases.TestCase.__call__` sans try/except wrappers.

File: pytest_django/plugin.py [hunk 1]
```diff
@@ -386,6 +386,20 @@ def _django_setup_unittest(request, django_db_blocker):
...
+                getattr(testMethod, "__unittest_skip__", False))
+
+            if not skipped:
+                self._pre_setup()
+            super(cls, self).debug()
+            if not skipped:
+                self._post_teardown()
+
+        cls.debug = _cleaning_debug
+
```
Comment (blueyed) [inline pytest_django/plugin.py]: Needs a comment/doc!

It was mentioned to report it for Django itself also.. has that been done?

**Comment by asfaltboy** [comment]
Django ticketed opened, for back ref: https://code.djangoproject.com/ticket/27391

**Comment by adamchainz** [comment]
@blueyed I didn't get around to opening the Django ticket as I said I would in pytest-dev/pytest#1977 ... thanks @asfaltboy !

**Comment by pelme** [comment]
Thanks for all debugging! I've merged this and will soon put out a 3.1.1 release which contains this fix!

**Pull request merged**